UnityProject/Assets/Scripts/Items/Implants/Organs/SlimeCore.cs

```diff
@@ -0,0 +1,222 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Chemistry;
+using Chemistry.Components;
+using HealthV2;
+using HealthV2.Living.PolymorphicSystems.Bodypart;
+using Systems.Character;
+using UI.CharacterCreator;
+using UnityEngine;
+
+[RequireComponent( typeof(ReagentCirculatedComponent))]
+public class SlimeCore : BodyPartFunctionality
+{
+
+
+	public int DropDownIndex = 0;
+
+	private CharacterSheet ForCore =>
+		new CharacterSheet()
+		{
+			Species = "SlimeBall",
+			SerialisedBodyPartCustom = new List<CustomisationStorage>()
+			{
+				new CustomisationStorage()
+				{
+					path = "/SlimeBody/SlimeCore" ,
+					Data = DropDownIndex + 1.ToString()
+				}
+
+			}
+		};
+
+	public SpriteDataSO AdultSprite;
+	public SpriteDataSO BabySprite;
+
+	private ReagentCirculatedComponent ReagentCirculatedComponent;
+
+
+	public List<ChanceAndPrefab> CanSplitInto = new List<ChanceAndPrefab>();
+
+
+	public Reagent SlimeJelly;
+
+
+	public float StartingAmount = 25;
+
+
+	public int CurrentNumberOfCore = 1;
+
+	public bool Stabilised = false;
+	public bool DeStabilised = false;
+
+
+	public bool Enhanced = false;
+	public bool EnhancedUsedUp = false;
+
+
+	public override void Awake()
+	{
+		base.Awake();
+		ReagentCirculatedComponent = this.GetComponentCustom<ReagentCirculatedComponent>();
+	}
+
+
+	[System.Serializable]
+	public class ChanceAndPrefab
+	{
+		[Range(0, 100)] public int ChanceToMutateTo;
+		public SlimeCore CoreMutateTo;
+	}
+
+	[NaughtyAttributes.Button()]
+
+	public void InitialiseBabySlime()
+	{
+		var Multiplier = (StartingAmount / ReagentCirculatedComponent.AssociatedSystem.BloodPool.Total);
+		ReagentCirculatedComponent.AssociatedSystem.BloodPool.Multiply(Multiplier);
+
+		foreach (var bodyPart in RelatedPart.HealthMaster.BodyPartList)
+		{
+			var Body =  bodyPart.GetComponent<SlimeBody>();
+
+			if (Body != null)
+			{
+				Body.BabySprite = BabySprite;
+				Body.AdultSprite = AdultSprite;
+				Body.CurrentState = null;
+			}
+
+		}
+	}
+
+	public override void OnAddedToBody(LivingHealthMasterBase livingHealth)
+	{
+
+		foreach (var bodyPart in livingHealth.BodyPartList)
+		{
+
+			var Body =  bodyPart.GetComponent<SlimeBody>();
+
+			if (Body != null)
+			{
+				Body.BabySprite = BabySprite;
+				Body.AdultSprite = AdultSprite;
+				Body.CurrentState = null;
+			}
+
+		}
+	} //Warning only add body parts do not remove body parts in this
+
+	[NaughtyAttributes.Button()]
+	public void DEBUGSlimesSplit()
+	{
+		var Multiplier =  (100f / ReagentCirculatedComponent.AssociatedSystem.BloodPool.Total);
+		ReagentCirculatedComponent.AssociatedSystem.BloodPool.Multiply(Multiplier);
+		SlimesSplit();
+	}
+
+	public bool CanSlimesSplit()
+	{
+		return ReagentCirculatedComponent.AssociatedSystem.BloodPool[SlimeJelly] > 99;
+	}
+
+	public void SlimesSplit()
+	{
+
+		if (CanSlimesSplit())
+		{
+
+			// Loop four times to create four offspring
+			for (int i = 0; i < 4; i++)
```

Comment:
should it be a random between 2 and 4?

Reply:
I think it's always 4?

Reply:
I think it's weird how it will constantly output 4 slimes no matter what, but its just my preference

Reply:
can be tweaked later on if needed